src/VisualStudio/CSharp/Impl/LanguageService/CSharpCreateServicesOnTextViewConnection.cs

```diff
@@ -39,6 +39,11 @@ internal class CSharpCreateServicesOnTextViewConnection : AbstractCreateServices
 
         protected override async Task InitializeServiceForOpenedDocumentAsync(Document document)
         {
+            // Preload project completion providers on a background thread since loading extensions can be slow
+            // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1488945
+            if (document.GetRequiredLanguageService<CompletionService>() is CompletionServiceWithProviders compeltionServiceWithProviders)
```

Comment:
i don't like the idea that this is a required service.  What about languages that don't support completion?

Reply:
This is inside a C# subtype (i.e. for CSharpContentType), so the service should always be available

Reply:
ah ok.